tasks/assemble.js

```diff
@@ -719,7 +719,19 @@ module.exports = function(grunt) {
    * @return {String}         The raw, assembled, uncompiled and unprocessed page
    */
   var injectBody = function(layout, body) {
-    return layout.replace(assemble.engine.bodyRegex, function () { return body; });
+    var indentation = getIndentation(layout);
+    var bodyWithIndentation = _.map(body.split('\n'), function (line){
+      return line.replace(/^/, indentation);
+    }).join('\n');
+    return layout.replace(assemble.engine.bodyRegex, function () { return bodyWithIndentation; });
+  };
+
+  var getIndentation = function(layout) {
+    var bodyRegex = assemble.engine.bodyRegex;
+    var bodyRegexWithIndentation = new RegExp(/\s*/.source + bodyRegex.source);
```

Comment:
shouldn't the whitespace matching pattern begin with `^`?  otherwise I think this would match all whitespace character in that line.